### ISSUE TITLE
NVJPEG failure: '#3' when trying to decode JPH file

### ISSUE THREAD

**Issue by jamesdavidson** [NVIDIA/CUDALibrarySamples#163]
Should I be able to open .jph file using nvJPEG2000 library?

The library docs advertise support for high throughput but so far I can only decode jp2 files.

>The [HTJ2K](https://htj2k.com) standard defines a wrapping file format known as JPH, which extends the JP2 file format with support for modern parameterizable colour spaces, images without any colour space, and custom colour mappings suitable for raw sensor compression.

I’m using Ubuntu 22.04 package nvjpeg2k0 0.7.5.32-1 amd64 with CUDA 11.5.

[original post in forums](https://forums.developer.nvidia.com/t/nvjpeg2000-decode-jph-file-format/275010)

**Comment by zohebk-nv** [comment]
JPH file format is not supported as of release v 0.7.5. We can look into adding support for this in an upcoming release.

**Comment by jamesdavidson** [comment]
If it is of interest then great! Let me know if you would like me to test out a draft / alpha version

**Comment by zohebk-nv** [comment]
Hi @jamesdavidson , would it be possible for you share a sample .jph file which causes nvjpeg2k to return an error?

**Comment by jamesdavidson** [comment]
Yep, here you go: https://gitlab.com/wg1/htj2k-codestreams/-/blob/master/testfiles_jph/file1_b11.jph

**Comment by jamesdavidson** [comment]
I now get NVJPEG failure: '#4' instead which is a more correct error message.

```
typedef enum
{
    NVJPEG2K_STATUS_SUCCESS                       = 0,
    NVJPEG2K_STATUS_NOT_INITIALIZED               = 1,
    NVJPEG2K_STATUS_INVALID_PARAMETER             = 2,
    NVJPEG2K_STATUS_BAD_JPEG                      = 3,
    NVJPEG2K_STATUS_JPEG_NOT_SUPPORTED            = 4,
    NVJPEG2K_STATUS_ALLOCATOR_FAILURE             = 5,
    NVJPEG2K_STATUS_EXECUTION_FAILED              = 6,
    NVJPEG2K_STATUS_ARCH_MISMATCH                 = 7,
    NVJPEG2K_STATUS_INTERNAL_ERROR                = 8,
    NVJPEG2K_STATUS_IMPLEMENTATION_NOT_SUPPORTED  = 9,
} nvjpeg2kStatus_t;
```

I'd still be interested in JPH file format support.

**Comment by zohebk-nv** [comment]
[v0.10.0 ](https://docs.nvidia.com/cuda/nvjpeg2000/releasenotes.html#v0-10-0) of nvjpeg2000 is able to support jph files along with magnitude refinement and singnificance propatagation passes required to decode the attached file. Please give this version a try and let me know if you see any issues

**Comment by jamesdavidson** [comment]
Thanks! That's great. No issues with [file1_b11.jph](https://gitlab.com/wg1/htj2k-codestreams/-/blob/master/testfiles_jph/file1_b11.jph). Much appreciated.
